### PR TITLE
fix `Signature` serialization and deserialization

### LINKED ISSUE #304: v3: serialize and deserialize `Signature` correctly
Currently, `Signature` is being serialized as: `{
  signature: {
    value: "12345"
  }
}`
Where as it should look like this: `{
  signature: "12345"
}`

This is a breaking change hence it will be part of v3.

### PR DESCRIPTION
**Pull request by mukulljangid**
fixes #304

### CODE REVIEW

File: xrpl4j-crypto-parent/xrpl4j-crypto-core/src/test/java/org/xrpl/xrpl4j/crypto/signing/SignatureTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    assertThat(signature).isEqualTo(deserialized);
+
+    WrappedSignature wrappedSignature = WrappedSignature.builder()
+      .signature(signature)
+      .build();
+    String serializedWrappedSignature = objectMapper.writeValueAsString(wrappedSignature);
+
+    String serialized = "{\"signature\":\"E2ACD61C90D93433402B1F704DA38DF72876B6788C2C05B3196E14BC711AECFF14A7D627643" +
+      "9A198D8B4880EE2DB544CF351A8CE231B3340F42F9BF1EDBF5104\"}";
+    assertThat(serializedWrappedSignature).isEqualTo(serialized);
```
Comment (mukulljangid) [inline xrpl4j-crypto-parent/xrpl4j-crypto-core/src/test/java/org/xrpl/xrpl4j/crypto/signing/SignatureTest.java:42]: Checkout this serialized object! @florent-uzio

Comment (florent-uzio) [reply xrpl4j-crypto-parent/xrpl4j-crypto-core/src/test/java/org/xrpl/xrpl4j/crypto/signing/SignatureTest.java:42]: Ah nice, thanks @mukulljangid , once this is merged and used in the cbdc-backend we will be able to merge our UI branch.
https://github.com/xpring-eng/cbdc-admin-ui/pull/323

File: xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/signing/Signature.java [hunk 2]
```diff
@@ -56,4 +69,27 @@ static ImmutableSignature.Builder builder() {
...
+   */
+  class SignatureDeserializer extends JsonDeserializer<Signature> {
+
+    @Override
+    public Signature deserialize(JsonParser jsonParser, DeserializationContext ctxt)
+      throws IOException {
+      JsonNode node = jsonParser.getCodec().readTree(jsonParser);
+      return Signature.builder().value(UnsignedByteArray.fromHex(node.asText())).build();
+    }
+  }
```
Comment (nkramer44) [inline xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/signing/Signature.java:94]: Let's put these in `xrpl4j-model` in the `jackson.modules` package

**Review by nkramer44 — CHANGES_REQUESTED** [review]
I think if we merge this into `main` it would technically introduce a breaking change, which I don't want to do. In the v3 branch this `Signature` is deprecated and replaced by `org.xrpl.xrpl4j.crypto.core.signing.Signature`. I think we should leave this `Signature` alone, switch over to the v3 branch and make the change on the new `Signature` class.

File: xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/signing/Signature.java (hunk 2 above)
Comment (mukulljangid) [reply xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/signing/Signature.java:94]: `Signature` is  a part of the `xrpl4j-crypto-core` module and might introduce circular dependency if we add crypto-core as a dependency in `xrpl4j-model` .

**Comment by mukulljangid** [comment]
Closing this. And opening a new PR which changes the model in v3 `Siganture`.

**Pull request closed without merging**